client/signup/steps/user/index.jsx

```diff
@@ -116,6 +120,13 @@ export class UserStep extends Component {
 		if ( oauth2Signup && clientId ) {
 			this.props.fetchOAuth2ClientData( clientId );
 		}
+		const experimentCheck = this.state.isDesktop
+			? 'registration_email_only_desktop'
+			: 'registration_email_only_mobile';
+
+		loadExperimentAssignment( experimentCheck ).then( ( experimentName ) => {
```

Comment:
Let's restrict this assignment only to `onboarding` flow users, otherwise we'll add to white noise by assigning other flow users to the experiment.

Reply:
@niranjan-uma-shankar ah, we've got an `onboarding` check in the `SignupForm` but that won't prevent user assignment. I'll move that check here, good call.